### ISSUE TITLE
Incorrect clang-format PointerAlignment for brace-enclosed initialization within if-statement

### ISSUE THREAD

**Issue by swalkner** [llvm/llvm-project#109371]
I can confirm this behaviour and would like to gently bump this report.

```cpp
if (MyClass * obj{GetMyClass()})
```
But - for some reason - `auto` and primitive-types seem to work as expected:
```cpp
if (auto* obj{GetMyClass()})
```
```cpp
if (int* obj{GetInt()})
```

I have reproduced that in clang-format:
* 17.0.3 (packaged with VS)
* 18.1.3 (and other versions via clang-format-configurator.site)

The only config option I changed:
> PointerAlignment: Left

(This issue has been split from #60146 )

**Comment by owenca** [comment]
Reverted by ee6468494e08651ec7f534781693408f34a74753 due to regressions.

**Comment by owenca** [comment]
`foo * bar{baz}` can be either a declaration or an expression depending on whether or not `foo` is a `TypeName`. Please use the `TypeNames` option to disambiguate it.
